flight/PiOS/inc/pios_can.h

```diff
@@ -35,6 +35,7 @@
 
 //! The set of CAN messages
 enum pios_can_messages {
+	PIOS_CAN_ERROR = -1,
```

Comment:
It's because there is a check somewhere else for something like `pios_can_messages foo; if(foo < 0)...`. Since the enum previously didn't support `<0`it would throw a warning that this condition always evaluates to false. Of course, this warning is wrong, since C does not prevent you from taking a variable `foo` and assigning any value you like to it, i.e. there's no GCC protection* that only enum values can be assigned to enum variables. 

Upshot: by making an error flag with a negative value, GCC no longer believes that the check is false. 

To be honest, I don't think it's a great solution. It's totally non-obvious why it works, but there doesn't seem to be another way to work around this GCC bug (as suggested on stackoverflow). 

[*] clang supports checking this just fine.